minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/command/StartRunner.java

```diff
@@ -279,15 +280,33 @@ private void regenerateFlowConfiguration(String flowConfigFileLocation) throws C
         }
     }
 
-    private Process startMiNiFi() throws IOException {
+    private Process startMiNiFi(BootstrapProperties bootstrapProperties) throws IOException {
+        int bootstrapListenPort = parseBootstrapListenPort(bootstrapProperties);
+
         MiNiFiListener listener = new MiNiFiListener();
-        listenPort = listener.start(runMiNiFi, bootstrapFileProvider, configurationChangeListener);
+        listenPort = listener.start(runMiNiFi, bootstrapListenPort, bootstrapFileProvider, configurationChangeListener);
 
         CMD_LOGGER.info("Starting Apache MiNiFi...");
 
         return startMiNiFiProcess(getProcessBuilder());
     }
 
+    private int parseBootstrapListenPort(BootstrapProperties bootstrapProperties) {
+        String bootstrapListenPort = bootstrapProperties.getProperty(NIFI_BOOTSTRAP_LISTEN_PORT);
+        int parsedBootstrapListenPort = Optional.ofNullable(bootstrapListenPort)
+            .map(String::trim)
+            .map(port -> {
+                try {
+                    return Integer.parseInt(port);
+                } catch (NumberFormatException e) {
+                    throw new IllegalArgumentException("Unable to parse bootstrap listen port, please provide a valid port for " + NIFI_BOOTSTRAP_LISTEN_PORT);
+                }
+            })
+            .orElse(0);
+        CMD_LOGGER.info("Using boostrap listen port {}", parsedBootstrapListenPort);
```

Comment:
Minor spelling correction, wording adjustment recommendation:
```suggestion
        CMD_LOGGER.info("Bootstrap listen port {}", parsedBootstrapListenPort);
```